packaging/linux/post_install.sh

```diff
@@ -37,6 +43,80 @@ make_mountpoint() {
   fi
 }
 
+systemd_exec_as() {
+    user=$1
+    shift
+
+    # Keybase supports running with and without systemd. In order for `su`
+    # to work with systemd commands, we need to pass in XDG_RUNTIME_DIR.
+    # /run/user/{uid} is the default when using systemd. We can't know the
+    # true value without installing systemd itself, which we don't want to
+    # depend on, so we use this default. Users who want to use a custom
+    # XDG_RUNTIME_DIR should specify it in their .profile so it is known
+    # by su -, which doesn't load pam_systemd. Unfortunately, this also
+    # results in an extra su invocation.
+    user_xdg_runtime_dir=""
+    # shellcheck disable=SC2016
+    # Intentionally do not expand $XDG_RUNTIME_DIR; we want the user's shell to expand it
+    if ! user_xdg_runtime_dir="$(su --login "$user" -c 'echo $XDG_RUNTIME_DIR')" || [ -z "$user_xdg_runtime_dir" ]; then
+        user_xdg_runtime_dir="/run/user/$(id -u "$user")" || return 1
+    fi
+
+    # To support restarting without systemd, we'd also have to pass in DISPLAY,
+    # but there's no easy way to figure that out.
+    su --login "$user" -c "XDG_RUNTIME_DIR=$user_xdg_runtime_dir $* 2> /dev/null"
+}
+
+# Exits with 0 iff the given user is running the service with systemd
+systemd_unit_active_for() {
+    user=$1
+    service=$2
+    command -v /usr/bin/systemctl &> /dev/null && systemd_exec_as "$user" "/usr/bin/systemctl --user -q is-active $service"
+}
+
+safe_restart_systemd_services() {
+    if ! autorestart_enabled; then
+        echo "Keybase autorestart disabled. Restart manually by running 'run_keybase' for each user using Keybase."
+        return 0
+    fi
+
+    while read -r pid; do
+        # Since keybase is running, we can assume run_keybase has been run before
+        # and the mountdir is configured (so, it is not a fresh install).
+        user="$(ps -o user= -p "$pid")"
+
+        restart_instructions="Aborting Keybase autorestart for $user. Restart manually by running 'run_keybase' as $user."
+
+        if ! systemd_unit_active_for "$user" "keybase.service"; then
+            echo "Keybase not running via systemd for $user."
+            echo "$restart_instructions"
```

Comment:
I think saying "Aborting Keybase autorestart" in the case where systemd isn't running is a bit aggressive, since it feels like an error when really the user configured it that way.  Can we soften it in the non-error cases to something like "Not auto-restarting Keybase..."?